src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/ManagedPal.Encrypt.cs

```diff
@@ -177,9 +178,36 @@ internal sealed partial class ManagedPkcsPal : PkcsPal
                 if (contentInfo.ContentType.Value == Oids.Pkcs7Data)
                 {
                     toEncrypt = EncodeOctetString(toEncrypt);
+                    return encryptor.OneShot(toEncrypt);
+                }
+                else
+                {
+                    using (MemoryStream memoryStream = new MemoryStream())
+                    {
+                        using (var cryptoStream = new CryptoStream(memoryStream, encryptor, CryptoStreamMode.Write))
```

Comment:
There's no need to drag CryptoStream and System.IO into this, because you only have one call to Write.  Just build the data and call encryptor.OneShot.

Reply:
I did it this way to avoid building intermediate buffers - this is a modified copy of OneShot method which tries to not unnecessarily allocate. I thought the purpose of CryptoStream was to feed it with data I want to encrypt/decrypt which I'm doing here.

To build data I'll also need some equivalent of MemoryStream so not sure what the ask is. Unless you mean something else?

Reply:
CryptoStream and MemoryStream both will create a fair amount of intermediate buffers.  But since the call to Write is in a loop I can believe that this code is easier to understand.

Reply:
Actually, it looks like this is just trying to do `reader.PeekContentBytes()`.  And I think Windows might not care that the outer tag is SEQUENCE.

Looks like this passes on Windows:

```C#
ContentInfo content = new ContentInfo(
    new Oid(Oids.Pkcs7Enveloped),
    "0403010203".HexToByteArray());

EnvelopedCms cms = new EnvelopedCms(content);

using (X509Certificate2 cert = Certificates.RSAKeyTransferCapi1.TryGetCertificateWithPrivateKey())
{
    cms.Encrypt(new CmsRecipient(cert));
    byte[] temp = cms.Encode();

    // Double checking.
    cms = new EnvelopedCms();
    cms.Decode(temp);
    cms.Decrypt(new X509Certificate2Collection(cert));
}

Assert.Equal("3003010203", cms.ContentInfo.Content.ByteArrayToHex());
```

I don't know that I agree with it passing, but since we're making a compat fix, we should be compatible.